File: api/src/main/java/com/doerapispring/domain/ListApplicationService.java
package com.doerapispring.domain;

import com.doerapispring.domain.events.TodoListEvent;
import scala.util.Try;

import java.util.List;
import java.util.function.BiFunction;
import java.util.function.Supplier;

public interface ListApplicationService {
    Try<TodoListModel> performOperation(
            User user,
            ListId listId,
            TodoListEvent event);

    TodoListModel getDefault(User user);

    CompletedTodoList getCompleted(User user, ListId listId);

    TodoListModel get(User user, ListId listId);

    List<TodoList> getAll(User user);

    void create(User user, String name);

    void setDefault(User user, ListId listId);
}

File: api/src/main/scala/com/doerapispring/domain/ListApplicationService.scala
package com.doerapispring.domain

import com.doerapispring.domain.events.TodoListEvent

trait ListApplicationService {
  def performOperation(user: User, listId: ListId, event: TodoListEvent): TodoListModel

  def getDefault(user: User): TodoListModel

  def getCompleted(user: User, listId: ListId): List[CompletedTodo]

  def get(user: User, listId: ListId): TodoListModel

  def getAll(user: User): List[TodoList]

  def create(user: User, name: String): Unit

  def setDefault(user: User, listId: ListId): Unit
}